src/vivarium/framework/resource/group.py

```diff
@@ -41,22 +44,31 @@ def __init__(
         """
         if not initialized_resources:
             raise ResourceError("Resource groups must have at least one resource.")
+        self._dependencies = dependencies
 
-        if len(set(r.component for r in initialized_resources)) != 1:
-            raise ResourceError("All initialized resources must have the same component.")
+        initialized_resources_ = (
+            [initialized_resources]
+            if isinstance(initialized_resources, Resource)
+            else initialized_resources
+        )
 
-        if len(set(r.resource_type for r in initialized_resources)) != 1:
+        # These checks are redundant since initialized_resources is either a list
+        # of Columns created by a single Component or a single Resource. We keep them
+        # in the event that changes in the future.
+        if len(set(res.component for res in initialized_resources_)) != 1:
```

Comment:
Is it worth keeping these around?

Reply:
I think it's good to keep them around, otherwise it looks weird how we're taking the component and type of the 0 index resource below. I don't think we need the comment above though?